fix: fix page header of web platform & status bar in More Action Page on iOS(OK-47963 OK-48007 OK-47970)

huhuanming opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a responsive TabPageHeader with collapsible sections, integrated search for small screens, notification indicator, and account selector support.

* **Bug Fixes / Behavior**
  * Unified page-type detection to improve modal/overlay handling across headers, pages, charts, and swap flows—leading to more consistent layout and interaction in overlay/modal contexts.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

huhuanming committed:
Create index.web.tsx

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/fb423de6-7e06-452e-9ba2-f5aa35015379?refs=85b5a09b0ac7ab77237fdd85f5fa11c7a1b95544&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/fb423de6-7e06-452e-9ba2-f5aa35015379/license?refs=85b5a09b0ac7ab77237fdd85f5fa11c7a1b95544&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

huhuanming committed:
useIsOverlayPage

huhuanming committed:
Add useIsModalPage hook and update usage in BasicPage

Introduces a new useIsModalPage hook in PageType HOC and replaces previous useIsOverlayPage usages with useIsModalPage in BasicPage.native.tsx to improve clarity and intent.

huhuanming committed:
Update BasicPage.native.tsx
